website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,218 @@
+---
+layout: post
+title:  "On Consuming (and Publishing) ES6+ Packages"
+author: Henry Zhu
+authorURL: https://twitter.com/left_pad
+date:   2018-06-25 12:00:00
+categories: announcements
+share_text: "On Consuming (and Publishing) ES6+ Packages"
+---
+
+How can we make compiling dependencies not just possible, but normal?
+
+<!--truncate-->
+
+This post is inspired by [Dan's tweet](https://twitter.com/dan_abramov/status/1009179550134296577) about how we can conflate shipping source code, ES6+, ESM
+
+This is a pretty enabling feature request for the whole ecosystem, so I'm glad we've tried to make this easier in Babel v7 (I just realized this whole post is a good pitch for using it). Hopefully it can be made more standard moving forward (if we're collectively able to figure out some things I outline below).
+
+## Why
+
+Why is compiling your dependencies desirable in the first place (vs. just your own code)?
+
+We can ship less code to users, since JavaScript has a [cost](https://medium.com/dev-channel/the-cost-of-javascript-84009f51e99e).
+
+## Assumptions
+
+- You can ship to modern browsers (don't have to support IE) or are able to send multiple types of bundles (one way being [checking for script`type=module`](https://philipwalton.com/articles/deploying-es2015-code-in-production-today/)).
+- And/or the dependencies actually publish ES6+ instead of ES5/ES3.
+
+## Backing up a Bit: Talking About `preset-env`
+
+If you think about it, the argument for why this would be helpful is the same for why we eventually introduced `@babel/preset-env`.
+
+Babel used to be `6to5`, since it only converted from ES6 to ES5. Back then, the browser support for ES6 was almost non-existent so the idea of a JS compiler was an amazing contribution to our ecosystem.
+
+I went through with implementing the preset even though there was so much complexity since it seemed to solve multiple issues for us. I didn't realize how much it would move us forward now that it's finally [surpassed `preset-es2015`](https://npm-stat.com/charts.html?package=babel-preset-es2015&package=babel-preset-env&from=2016-11-01&to=2018-06-22) which is a nice accomplishment.
+
+When I think about Babel's role not just in moving the language itself forward but helping us develop websites better, I saw that we would eventually want to move past only compiling to ES5. Creating `preset-env` helps us align with the browsers since if we only compiled to ES5, no one would ever run native code in the browsers.
+
+The real difference is realizing that there will _always_ be a sliding window of support: whether it's the JavaScript language that continues to move forward with new syntax, the browsers catching up with the standard, or the companies/projects themselves.
```

Comment:
I gave a talk proposing shipping progressive levels of transpilation in 2015, and by far the most positive feedback I got from it was around my section addressing this. 
That we have to come up with a forward-safe paradigm because the world and web are always pushing forward. Shifting the mindset from “this is my target” to “this is my target for now”.